### PR TITLE
Thomas/fuzzy match functions threshold

### PR DESCRIPTION
**Pull request by balzdur**
Final steps to fully implement fuzzy match edit view :
- add threshold edit
- refactor code to extract state logic in a separate file

### CODE REVIEW

File: packages/app-builder/src/models/fuzzy-match.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+export type FuzzyMatchAlgorithm = (typeof fuzzyMatchAlgorithms)[number];
+
+export function isFuzzyMatchAlgorithm(
+  value: string,
+): value is FuzzyMatchAlgorithm {
+  return (fuzzyMatchAlgorithms as ReadonlyArray<string>).includes(value);
+}
+
+export function getFuzzyMatchAlgorithmName(
+  t: TFunction<['common', 'scenarios'], undefined>,
```
Comment (Pascal-Delange) [inline packages/app-builder/src/models/fuzzy-match.ts:21]: in the end t is not used, right ?

Comment (Pascal-Delange) [reply packages/app-builder/src/models/fuzzy-match.ts:21]: don't we want to store those in the translation json files ?

Comment (Pascal-Delange) [reply packages/app-builder/src/models/fuzzy-match.ts:21]: (this is where we could use "bag of words similarity" and "full string similarity", or something like this, for more human readable names)

File: packages/app-builder/src/components/Scenario/AstBuilder/AstBuilderNode/FuzzyMatchComparatorEdit/Modal.tsx [hunk 2]
```diff
@@ -300,3 +225,75 @@ function FuzzyMatchComparatorEditModalContent({
...
+
+const funcNameTKeys = {
+  FuzzyMatch: 'scenarios:edit_fuzzy_match.fuzzy_match',
+  FuzzyMatchAnyOf: 'scenarios:edit_fuzzy_match.fuzzy_match_any_of',
+} satisfies Record<
+  FuzzyMatchEditorNodeViewModel['funcName'],
+  ParseKeys<['scenarios']>
+>;
+
+function Examples({
```
Comment (Pascal-Delange) [inline packages/app-builder/src/components/Scenario/AstBuilder/AstBuilderNode/FuzzyMatchComparatorEdit/Modal.tsx:237]: This is really pretty cool 👌  great idea on this

**Review by Pascal-Delange — APPROVED** [review]
cool!

File: packages/app-builder/src/models/fuzzy-match.ts (hunk 1 above)
Comment (balzdur) [reply packages/app-builder/src/models/fuzzy-match.ts:21]: I added translation for both editable algorithm.
Just check I did not associated wrong translation to each algorithm, and if it's OK, this is mergeable.

Comment (Pascal-Delange) [reply packages/app-builder/src/models/fuzzy-match.ts:21]: looks good !

**Pull request merged**